yesod-form/Yesod/Form/Functions.hs

```diff
@@ -336,21 +336,24 @@ identifyForm identVal form = \fragment -> do
     -- Create hidden <input>.
     let fragment' =
           [shamlet|
-            <input type=hidden name=#{identifyFormKey} value=#{identVal}>
+            <input type=hidden name=#{identifyFormKey} value=identify-#{identVal}>
             #{fragment}
           |]
 
     -- Check if we got its value back.
     mp <- askParams
-    let missing = (mp >>= Map.lookup identifyFormKey) /= Just [identVal]
+    let missing = (mp >>= Map.lookup identifyFormKey) /= Just ["identify-" <> identVal]
 
     -- Run the form proper (with our hidden <input>).  If the
     -- data is missing, then do not provide any params to the
     -- form, which will turn its result into FormMissing.  Also,
     -- doing this avoids having lots of fields with red errors.
     let eraseParams | missing   = local (\(_, h, l) -> (Nothing, h, l))
                     | otherwise = id
-    eraseParams (form fragment')
+    ( res', w) <- eraseParams (form fragment')
+
+    let res = if missing then FormMissing else res'
```

Comment:
To ensure this doesn't get reverted in the future: can you add comment linking to the issue?